posts/posts-router.js

```diff
@@ -0,0 +1,118 @@
+const express = require('express');
+
+const Hubs = require('./db.js');
+
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+    try {
+        const hubs = await Hubs.find();
+        res.status(200).json(hubs);
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'The posts information could not be retrieved.',
+        });
+    }
+});
+
+
+router.get('/:id', async (req, res) => {
+    try {
+      const hub = await Hubs.findById(req.params.id);
+  
+      if (hub) {
+        res.status(200).json(hub);
+      } else {
+        res.status(404).json({ message: 'Hub not found' });
```

Comment:
Here I'd recommend to check what the async call is returning in case there is no hub in the database, according to the documentation given in the project, when this happens you should get an empty array, but empty arrays are still true(different than empty strings which are false). So your if/ else statement should always evaluate to true the way you have it written. You can check instead for array.length, if its 0 then you'll get the falsy value.